src/processes.c

```diff
@@ -778,6 +820,89 @@ static void ps_submit_fork_rate (derive_t value)
 
 /* ------- additional functions for KERNEL_LINUX/HAVE_THREAD_INFO ------- */
 #if KERNEL_LINUX
+static procstat_t *ps_read_tasks_status (int pid, procstat_t *ps)
+{
+	char           dirname[64];
+	DIR           *dh;
+	char           filename[64];
+	FILE          *fh;
+	struct dirent *ent;
+	unsigned long long cswitch_vol = 0;
+	unsigned long long cswitch_invol = 0;
+	char buffer[1024];
+	char *fields[8];
+	int numfields;
+
+	ssnprintf (dirname, sizeof (dirname), "/proc/%i/task", pid);
+
+	if ((dh = opendir (dirname)) == NULL)
+	{
+		DEBUG ("Failed to open directory `%s'", dirname);
+		return (NULL);
+	}
+
+	while ((ent = readdir (dh)) != NULL)
+	{
+		char *tpid;
+
+		if (!isdigit ((int) ent->d_name[0]))
+			continue;
+
+		tpid = ent->d_name;
+
+		ssnprintf (filename, sizeof (filename), "/proc/%i/task/%s/status", pid, tpid);
+		if ((fh = fopen (filename, "r")) == NULL)
+		{
+			DEBUG ("Failed to open file `%s'", filename);
+			continue;
+		}
+
+		while (fgets (buffer, sizeof(buffer), fh) != NULL)
+		{
+			long long tmp;
```

Comment:
We have:

* `long long tmp`
* `unsigned long long cswitch_vol`
* `derive_t ps->cswitch_vol`

Can you please use `derive_t` for all of these?